cohortextractor/study_definition.py

```diff
@@ -70,6 +73,15 @@ def to_file(
             df["patient_id"] = default_rng().choice(
                 (len(df) * 10), size=len(df), replace=False
             )
+            if not is_csv_filename(filename):
+                # Convert the data generated by make_df_from_expectations to the correct
+                # format for binary outputs.  This matches the approach taken in the
+                # backend .to_file methods.
+                buffer = StringIO()
```

Comment:
Users sometimes hit memory issues when generating large dummy cohorts locally (particularly as Docker on mac/Win runs in a VM with a fixed memory allocation). This will approach will exacerbate that problem, but I think that probably the right thing to do is just to roll with it for now and address user issues if/when they arise.